### PR TITLE
Creating compatibility with Kubeflow Dashboard UI

### LINKED ISSUE #12657: [BUG] Unexpected token '%' - Kubernetes Cluster - integrated into Kubeflow UI
### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

2.14.1

### System information

- Kubernetes Cluster (various types)
- Integrated into the Kubeflow UI (as Iframe)

### Describe the problem

Kubeflow integrates features into its UI via IFrames, which again re-encodes the URI. MLflow cannot deal with it (as of now).

withErrorBoundary.tsx:29 Caught Unexpected Error:  SyntaxError: Unexpected token '%', "[%2248f495a"... is not valid JSON
 
SyntaxError: Unexpected token '%', "[%2248f495a"... is not valid JSON (CompareRunPage.tsx)
 
Utils.tsx:994 Could not parse experiment query parameter %5B%220%22%2C%221%22%5D 

These are typical errors getting thrown in the console. I have found this error in the compare run feature and sort run feature on experiment overview.

### Steps to reproduce the bug

Install MLflow into Kubeflow

Create Some Runs

Try to compare runs or sort runs.

### Code to generate data required to reproduce the bug

Generic MLflow Runs

### Is the console panel in DevTools showing errors relevant to the bug?

react-dom.production.min.js:189 SyntaxError: Unexpected token '%', "[%2248f495a"... is not valid JSON
    at JSON.parse (<anonymous>)
    at Function.mapToProps (CompareRunPage.tsx:70:25)
    at r (wrapMapToProps.js:41:46)
    at r.mapToProps (wrapMapToProps.js:50:19)
    at r (wrapMapToProps.js:41:46)
    at selectorFactory.js:23:18
    at connectAdvanced.js:327:16
    at Object.useMemo (react-dom.production.min.js:184:189)
    at t.useMemo (react.production.min.js:25:208)
    at v (connectAdvanced.js:312:30)

### Does the network panel in DevTools contain failed requests relevant to the bug?

localhost:5000/mlflow/?runs=%5B%252281d708375e574d6cbf4985b8701d67d2%2522,%25225f70fea1ef004d3180a6c34fe2d0d94e%2522%5D&experiments=%5B%25220%2522%5D

### PR DESCRIPTION
**Pull request by cgilviadee**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/cgilviadee/mlflow/pull/12663?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12663/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12663
```

</p>
</details>

A lot of projects nowadays are combining MLflow and Kubeflow to get the best of both worlds. When integrating MLflow into the Kubeflow dashboard there is a bug that breaks the compare runs feature. The log says  "Unexpected token '%', "[%2248f495a"... is not valid JSON". I guess the culprit behind this is the iframe. The paramters get re-encoded (sometimes multiple times over). I am not too sure if this is the right place to put it, but it fixes the bug for me.

### Related Issues/PRs

Resolve  #12657

### What changes are proposed in this pull request?

Fix "Unexpected token '%', "[%2248f495a"... is not a valid JSON" error. Making MLflow more robust.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. 

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/server/js/src/experiment-tracking/components/CompareRunPage.tsx [hunk 1]
```diff
@@ -63,9 +63,18 @@ class CompareRunPageImpl extends Component<CompareRunPageProps> {
   }
 }
 
+const decodeURI = (uri: string): string => {
```
Comment (daniellok-db) [inline mlflow/server/js/src/experiment-tracking/components/CompareRunPage.tsx:66]: could we add a comment here describing why this is necessary?

**Review by daniellok-db — APPROVED** [review]
overall looks good to me! let's wait for tests to pass

**Comment by daniellok-db** [comment]
looks like you'll need to run `yarn prettier:fix` in `mlflow/server/js` to resolve the lint error

**Comment by cgilviadee** [comment]
Done.

**Comment by B-Step62** [comment]
All failed tests are due to the breaking change in setfit and unrelated to this PR. We've added a workaround in master https://github.com/mlflow/mlflow/pull/12709.

**Comment by B-Step62** [comment]
Thank you so much for the contribution, @cgilviadee!!

**Pull request merged**